src/App.tsx

```diff
@@ -63,32 +65,63 @@ const App = (): JSX.Element => {
     }
   }, [notifications]);
 
-  const getureTimeAgo = useRef(new Date());
+  const [scaleState, setScaleState] = useState<{
+    visible: boolean;
+    size: 'small' | 'full';
+  }>({ visible: false, size: 'small' });
+
+  const isQuickScaleVisible = scaleState.visible && scaleState.size === 'small';
+  useEffect(() => {
+    if (!isQuickScaleVisible) return;
+
+    const scheduleHide = () =>
+      setTimeout(() => {
+        setScaleState((state) => ({ ...state, visible: false }));
+      }, 10000);
+    let timer = scheduleHide();
+
+    let lastSignificantWeight = store.getState().stats.sensors.w;
+    const subscription = store.subscribe(() => {
+      const weight = store.getState().stats.sensors.w;
+      if (Math.abs(weight - lastSignificantWeight) > 2) {
+        lastSignificantWeight = weight;
+        clearTimeout(timer);
+        timer = scheduleHide();
+      }
+    });
+
+    return () => {
+      clearTimeout(timer);
+      subscription();
+    };
+  }, [isQuickScaleVisible]);
```

Comment:
I assume this wont be the last overlay we want to build. Would it make sense to generalize?